ffi/utf8proc.lua

```diff
@@ -30,4 +30,16 @@ function Utf8Proc.lowercase(str)
     return folded_str
 end
 
+function Utf8Proc.strlen(str)
+    local str_buf = ffi.cast("const utf8proc_uint8_t *", str)
+    local buffer = ffi.new("utf8proc_int32_t[1]")
+    local retval = 0
+    local pos = 0
+    while str_buf[pos] ~= 0 do
+        pos = pos + libutf8proc.utf8proc_iterate(str_buf + pos, -1, buffer)
+        retval = retval + 1
```

Comment:
May be make this easier to read by switching lines 39 and 40:
```
while char at pos is not NULL:
   count = count + 1
   advance pos
```
Also, name retval "count" or something, we know it's what will be returned with `return count` - `return retval` is redundant :)